src/api/routes/offer.js

```diff
@@ -28,6 +28,38 @@ module.exports = (app) => {
         }
     });
 
+    /**
+     * Gets an offer from the database with its id
+    */
+    router.get("/:offer", validators.get, async (req, res, next) => {
+        try {
+            const offer = await (new OfferService()).getOfferByID({ id: req.params.offer });
+
+            // Later add here requirements
+            const filters = [
+                "title",
+                "jobStartDate",
+                "jobMinDuration",
+                "jobMaxDuration",
+                "description",
+                "vacancies",
+                "isPaid",
+                "location",
+                "jobType"
+            ];
+
+            const filteredOffer = {};
+
+            filters.forEach((key) => {
+                filteredOffer[key] = offer[key];
+            });
```

Comment:
If you are fetching an offer by its id, don't bother having filters as it will be redundant

Reply:
So, just return the actual offer object found in the DB?

Reply:
Exactly, and return an "error response" in case of error/not found